Add vlan validation in config interface ip add command

matiAlfaro opened the pull request:
~

<!--
    Please make sure you've read and understood our contributing guidelines:
    https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

    ** Make sure all your commits include a signature generated with `git commit -s` **

    If this is a bug fix, make sure your description includes "closes #xxxx",
    "fixes #xxxx" or "resolves #xxxx" so that GitHub automatically closes the related
    issue when the PR is merged.

    If you are adding/modifying/removing any command or utility script, please also
    make sure to add/modify/remove any unit tests from the tests
    directory as appropriate.

    If you are modifying or removing an existing 'show', 'config' or 'sonic-clear'
    subcommand, or you are adding a new subcommand, please make sure you also
    update the Command Line Reference Guide (doc/Command-Reference.md) to reflect
    your changes.

    Please provide the following information:
-->

#### What I did
Fix for -16975
Prevent configuration of illegal Vlan name in "config interface ip add" command
#### How I did it
Added Vlan name validation
#### How to verify it
##### Positive tests
config interface ip add Vlan100 1.1.1.1/24
config interface ip add Vlan4093 1.1.1.1/24
##### Negative tests
config interface ip add Vlan000000000000003 1.1.1.1/24
config interface ip add Vlan01 1.1.1.1/24
config interface ip add Vlan1.2 1.1.1.1/24
#### Previous command output (if the output of a command-line utility has changed)
On illegal Vlan name command succeeded with no error
#### New command output (if the output of a command-line utility has changed)
config interface ip add Vlan030 1.1.1.1/24
Error: Vlan030 is not a valid Vlan name

Commit:
Add vlan validation in config interface ip add command

~

Signed-off-by: mati <mati@marvell.com>

amazor commented on config/main.py:
```diff
@@ -4544,6 +4551,7 @@ def add(ctx, interface_name, ip_addr, gw):
     if interface_is_in_portchannel(portchannel_member_table, interface_name):
         ctx.fail("{} is configured as a member of portchannel."
                 .format(interface_name))
+    
```
Remove extra line

amazor commented on config/main.py:
```diff
@@ -4513,6 +4513,13 @@ def ip(ctx):
+def validate_vlan_format(text):
+    pattern = re.compile(r'^Vlan([1-9]\d{0,2}|[1-3]\d{3}|40[0-8][0-9]|409[0-4])$')
+
+    if pattern.match(text):
```
_return pattern.fullmatch(text) is not None_

I would use fullmatch instead of match, and remove the '^' and '$' (for simplicity's sake). because match returns an object only if it matches from the beginning of the string.
ex:
re.match('pattern', 'pattern in string') will match
re.fullmatch('pattern', 'pattern in string') will NOT match

matiAlfaro replied:
Thanks

amazor commented on config/main.py:
```diff
@@ -4513,6 +4513,13 @@ def ip(ctx):
     pass
 
+def validate_vlan_format(text):
+    pattern = re.compile(r'^Vlan([1-9]\d{0,2}|[1-3]\d{3}|40[0-8][0-9]|409[0-4])$')
```
This is a pretty complicated regex, make sure to test this with multiple different valid (and invalid) strings if you haven't already.
I recommend https://regexr.com and copying some vlan strings to validate this expression works as expected.

matiAlfaro replied:
Thanks :-)

shiraez commented on config/main.py:
```diff
@@ -4513,6 +4513,13 @@ def ip(ctx):
+    if pattern.match(text):
+        return True
+    else:
+        return False
```
You can change it to:
```python
return bool(pattern.match(text))
```

Commit:
review fixes & add tests

Signed-off-by: mati <mati@marvell.com>

Commit:
Review fixes - Vlan exists

Test that vlan exists
Added an optional param (option) --ignore-vlan which will test only
format

~

Signed-off-by: matiAlfaro <mati@marvell.com>

Commit:
Return error if vlan doesn't exist

Signed-off-by: matiAlfaro <mati@marvell.com>
